src/lib/u2/ViewFactory.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+foam.CLASS({
+  package: 'foam.u2',
+  name: 'ViewFactory',
+  extends: 'foam.core.Function',
+
+  properties: [
+      // foam.core.Function checks that the type is function; be more flexible.
+    [ 'assertValue', function() { } ]
+  ],
+
+  methods: [
+    function installInProto(proto) {
+      this.SUPER(proto);
+
+      var name = this.name;
+      var oldRaw;
+      var cached;
+
+      Object.defineProperty(proto, name + '$f', {
```

Comment:
I guess that could cause problems if someone takes the $f value and passes it around, it would need to be bound to the object like a listener